From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Toffikk <slodkitofik@gmail.com>
Date: Sat, 20 Dec 2025 23:19:08 +0100
Subject: [PATCH] Appreciate base patches more

This is an example base patch.

To generate one like this you generally follow the same steps as creating feature patches but with a small caveat; for it to *actually* be rebuilt and saved as a base patch it HAS to be moved *BEFORE* the `baguette Base Patches` commit, which is used internaly as an empty marker commit which seperates the base patches from per-file patches.
An example way of adding a base patch would be:
1. Make your changes
2. Do `git commit` with your commit message
3. Use `git rebase -i base` and move the commit you just created before the `Base Patches` commit
4. Follow with `git rebase --continue`
5. And lastly rebuild))

Modifying already present base patches works the same way as modifying feature patches: by editing the relevant commit
Base patches benefit mostly forks which would for example want to pull other forks, for that they can use the base patches which are applied before the fork's file patches (and therefore features) to prepare a base for their fork to work on which would be nearly identical to just making that fork your primary upstream and that's just one of its benefits!

They can also be used the same way as in the upstream project (canvas) for involving changes that you wish to be made before your actual code changes so you can use per-file patches safely without worrying about patch clutter from changes that modify hundreds of files.

diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 4d23765443614b1e55dbde1f9ac5250e4e30e12f..be3f62357e46e5a4cbabd315dd07d672a211eb34 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1314,6 +1314,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             // Paper start - Add onboarding message for initial server start
             if (io.papermc.paper.configuration.GlobalConfiguration.isFirstStart) {
                 LOGGER.info("*************************************************************************************");
+                LOGGER.info("Hey base patch enjoyer!"); // Baguette - appreciate base patches more
                 LOGGER.info("This is the first time you're starting this server.");
                 LOGGER.info("It's recommended you read our 'Getting Started' documentation for guidance.");
                 LOGGER.info("View this and more helpful information here: https://docs.papermc.io/paper/next-steps");
